core/src/main/java/org/apache/iceberg/actions/FileRewriteGroup.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.List;
+import org.apache.iceberg.ContentFile;
+import org.apache.iceberg.ContentScanTask;
+
+/**
+ * Container class representing a set of files to be rewritten by a {@link FileRewriteExecutor}.
+ *
+ * @param <I> the Java type of the plan info member like {@link RewriteDataFiles.FileGroupInfo} or
+ *     {@link RewritePositionDeleteFiles.FileGroupInfo}
+ * @param <T> the Java type of the tasks to read the files which are rewritten
+ * @param <F> the Java type of the content files which are rewritten
+ */
+public abstract class FileRewriteGroup<I, T extends ContentScanTask<F>, F extends ContentFile<F>> {
```

Comment:
it is confusing to introduce `FileRewriteGroup` name with another existing class named `RewriteFileGroup`.

We can probably call this `BaseFileGroup` or `BaseRewriteFileGroup`.

Reply:
Names staring with `Base` are reserved for the immutable interfaces for the actions themselves.

I would like to keep the FileRewriteGroup/FileRewritePlan/FileRewritePlanner/FileRewriteRunner similar and/or grouped together.

Maybe create a new package (org.apache.iceberg.actions.rewrite)? Maybe a name starting with the same prefix (RewriteGroup/RewritePlan/RewritePlanner/RewriteRunner)?

Reply:
> Names staring with Base are reserved for the immutable interfaces for the actions themselves.

Is `Base*` reserved only for actions in that package? Personally, I don't see an issue of adding classes like `BaseRewriteFileGroup`, `BaseRewriteFilePlan` etc.

Reply:
Currently only action base classes are there with name starting with base.

`RewriteFilePlan` will be the final class. I don't plan to add different plans for different compaction methods. They will only differ in the `RewriteFileGroup`s.

What about the planner/runner interfaces? They are definitely not `base`